functions/functionsv2/slack/slack_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slack
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/api/kgsearch/v1"
+	"google.golang.org/api/option"
+)
+
+var slackURL string
+
+// TestMain sets up the config rather than using the config file
+// which contains placeholder values.
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+	slackURL = os.Getenv("GOLANG_SAMPLES_SLACK_URL")
+	kgKey = os.Getenv("GOLANG_SAMPLES_KG_KEY")
+	if kgKey == "" {
+		log.Print("GOLANG_SAMPLES_KG_KEY is unset. Skipping.")
```

Comment:
Should this be a `warning` or an `error`?

(I'd imagine the latter is easier to detect via CI.)

Reply:
There is no way (besides Exit()) to error out during TestMain. It would be nice to refactor this a bit, but i think that's out of scope for now.